plutus-tx-plugin/test/Budget/Spec.hs

```diff
@@ -39,165 +40,212 @@ tests = testNestedGhc "Budget" [
 
   , goldenBudget "anyExpensive" compiledAnyExpensive
   , goldenUPlcReadable "anyExpensive" compiledAnyExpensive
   , goldenPirReadable "anyExpensive" compiledAnyExpensive
+  , goldenEvalCekCatch "anyExpensive-result" [compiledAnyExpensive]
 
   , goldenBudget "anyEmptyList" compiledAnyEmptyList
   , goldenUPlcReadable "anyEmptyList" compiledAnyEmptyList
   , goldenPirReadable "anyEmptyList" compiledAnyEmptyList
+  , goldenEvalCekCatch "anyEmptyList-result" [compiledAnyEmptyList]
 
   , goldenBudget "allCheap" compiledAllCheap
   , goldenUPlcReadable "allCheap" compiledAllCheap
   , goldenPirReadable "allCheap" compiledAllCheap
+  , goldenEvalCekCatch "allCheap-result" [compiledAllCheap]
 
   , goldenBudget "allExpensive" compiledAllExpensive
   , goldenUPlcReadable "allExpensive" compiledAllExpensive
   , goldenPirReadable "allExpensive" compiledAllExpensive
+  , goldenEvalCekCatch "allExpensive-result" [compiledAllExpensive]
 
   , goldenBudget "allEmptyList" compiledAllEmptyList
   , goldenUPlcReadable "allEmptyList" compiledAllEmptyList
   , goldenPirReadable "allEmptyList" compiledAllEmptyList
+  , goldenEvalCekCatch "allEmptyList-result" [compiledAllEmptyList]
 
   , goldenBudget "findCheap" compiledFindCheap
   , goldenUPlcReadable "findCheap" compiledFindCheap
   , goldenPirReadable "findCheap" compiledFindCheap
+  , goldenEvalCekCatch "findCheap-result" [compiledFindCheap]
 
   , goldenBudget "findExpensive" compiledFindExpensive
   , goldenUPlcReadable "findExpensive" compiledFindExpensive
   , goldenPirReadable "findExpensive" compiledFindExpensive
+  , goldenEvalCekCatch "findExpensive-result" [compiledFindExpensive]
 
   , goldenBudget "findEmptyList" compiledFindEmptyList
   , goldenUPlcReadable "findEmptyList" compiledFindEmptyList
   , goldenPirReadable "findEmptyList" compiledFindEmptyList
+  , goldenEvalCekCatch "findEmptyList-result" [compiledFindEmptyList]
 
   , goldenBudget "findIndexCheap" compiledFindIndexCheap
   , goldenUPlcReadable "findIndexCheap" compiledFindIndexCheap
   , goldenPirReadable "findIndexCheap" compiledFindIndexCheap
+  , goldenEvalCekCatch "findIndexCheap-result" [compiledFindIndexCheap]
 
   , goldenBudget "findIndexExpensive" compiledFindIndexExpensive
   , goldenUPlcReadable "findIndexExpensive" compiledFindIndexExpensive
   , goldenPirReadable "findIndexExpensive" compiledFindIndexExpensive
+  , goldenEvalCekCatch "findIndexExpensive-result" [compiledFindIndexExpensive]
 
   , goldenBudget "findIndexEmptyList" compiledFindIndexEmptyList
   , goldenUPlcReadable "findIndexEmptyList" compiledFindIndexEmptyList
   , goldenPirReadable "findIndexEmptyList" compiledFindIndexEmptyList
+  , goldenEvalCekCatch "findIndexEmptyList-result" [compiledFindIndexEmptyList]
 
   , goldenBudget "filter" compiledFilter
   , goldenUPlcReadable "filter" compiledFilter
   , goldenPirReadable "filter" compiledFilter
+  , goldenEvalCekCatch "filter-result" [compiledFilter]
```

Comment:
That is getting repetitive, how about defining a function encompassing the four types of tests? We'll probably need it in other places too.

Reply:
Yeah, I have also wondered about e.g. just combining the result value and the budget into a single file...